include/revng/Pipeline/Analysis.h

```diff
@@ -53,6 +53,12 @@ class AnalysisWrapperImpl : public AnalysisWrapperBase {
     Invokable.dump(OS, Indentation);
   }
 
+  llvm::Error
+  precondition(const Context &Ctx,
+               const ContainerToTargetsMap &Containers) const override {
+    return Invokable.precondition(Ctx, Containers);
+  }
+
```

Comment:
Preconditions are only for pipes, which cannot fail.
Analysis can fail, they don't need this. Drop drop drop.